rust-runtime/aws-smithy-xml/src/decode.rs

```diff
@@ -256,6 +254,13 @@ impl<'inp> Document<'inp> {
     }
 }
 
+/// A new-type wrapper around `Token` to prevent the wrapped third party type from showing up in
+/// public API
+#[derive(Debug)]
+pub struct XmlToken<'inp> {
+    token: Token<'inp>,
+}
```

Comment:
Nit: might be slightly easier to read as `XmlToken<'inp>(Token<'inp>)`

Reply:
Agreed. Addressed in 3105f8f.